rust/chains/hyperlane-ethereum/src/contracts/multicall.rs

```diff
@@ -34,16 +41,40 @@ pub async fn build_multicall<M: Middleware + 'static>(
     Ok(multicall)
 }
 
-pub fn batch<M: Middleware, D: Detokenize>(
+pub async fn batch<M, D>(
     multicall: &mut Multicall<M>,
     calls: Vec<ContractCall<M, D>>,
-) -> ContractCall<M, Vec<MulticallResult>> {
+) -> ContractCall<M, Vec<MulticallResult>>
+where
+    M: Middleware + 'static,
+    D: Detokenize,
+{
     // clear any calls that were in the multicall beforehand
     multicall.clear_calls();
 
+    let mut individual_estimates_sum = Some(U256::zero());
+    let overhead_per_call: U256 = MULTICALL_OVERHEAD_PER_CALL.into();
+
     calls.into_iter().for_each(|call| {
+        individual_estimates_sum = match call.tx.gas() {
+            Some(gas_estimate) => {
+                let gas_estimate: U256 = gas_estimate.into();
+                individual_estimates_sum.map(|sum| sum + gas_estimate + overhead_per_call)
+            }
+            None => {
+                warn!(call=?call.tx, "Failed to estimate gas for call in batch");
```

Comment:
curious if you think it's worth just estimating here if this happens?
I'm happy either way. I think the log message / comment below are stale -- maybe something like "unknown gas limit for batched call, falling back to estimating gas for entire batch"